### ISSUE TITLE
Start in private mode and handle user account automatically

### ISSUE THREAD

**Issue by xPaw** [thelounge/electron-lounge#1]
It needs to create an account, and automatically login into it when the app starts. Log out button also has to be hidden.

**Comment by MaxLeiter** [comment]
It would be kind of cool to be able to point it to a running Lounge instance, but maybe not worth it. That's how I made the TUI POC work.

**Comment by lcarva** [comment]
I gave this a go by storing a randomly generated password in native password storage via [keytar](https://github.com/atom/node-keytar), and creating a local user.

I'm stuck in getting the username and password values injected into the sign-in form. From what I'm reading I don't think accessing the DOM (in renderer process) is possible from index.js (main process).

Any ideas?

**Comment by lcarva** [comment]
I'm wondering if:

1. Not set LOUNGE_HOME; and simply use default value; this will make more intuitive to create/manage users as needed via `lounge add <user>`.
2. Upon electron start up (before window is created), check if at least one user exists. If so, carry on. Otherwise, display a [message box](https://github.com/electron/electron/blob/master/docs/api/dialog.md#dialogshowmessageboxbrowserwindow-options-callback) indicating that a user must be first be created.

**Comment by MaxLeiter** [comment]
@lcarva yeah, one of the largest issues with this setup is electron <=> frontend communictaion. Check out https://github.com/electron/electron/blob/master/docs/api/ipc-main.md for documentation on how-to.

**Comment by xPaw** [comment]
@lcarva Yeah that's a problem. It looks like localStorage can't be accessed from main process either.

**Comment by lcarva** [comment]
Right. I'm not well versed in electron, but it seems that automatic login as requested here is possible only with changes in thelounge app itself. Given the still small popularity of this electron wrapper, that's probably not something we want. 

Any thoughts on my suggestion above regarding explicit user management? It doesn't provide the automatic login UX, but it does seem like a way to address the security concern in the mean time.

**Comment by xPaw** [comment]
Eh, that's a pretty silly solution for standalone, but still better than having hardcoded credentials. So let's do that at least for now.

**Comment by lcarva** [comment]
We can reach the DOM by using a preload script with ipc-main. I can make the renderer request credentials, have the main process return this value, then the renderer populates the input elements and presses "sign in". User is logged in automatically.

The hiccup now is determining when to start the login process. ATM I have a function on setTimeout which is unacceptable. Looking at the lounge source code, I don't see a DOM event I can bind to. TBH I'm not sure what kind of framework is used for the client portion of the code. Ideally, there would be an "auth-requested" event, or even "sign-in" page loaded event.

**Comment by MaxLeiter** [comment]
@lcarva any progress on this front?

**Comment by lcarva** [comment]
@MaxLeiter, no, no progress. I haven't had the time to further investigate this.

My older self also has hesitations with the designed approach. Generating and injecting credentials automatically seems error prone and insecure. IMO this completely outweighs the benefit of automatic login. Especially, because generating the user and performing login is a one time operation. The Lounge does a pretty good job at saving the login token and doing the auto login for subsequent sessions. IOW, I've only ever had to login manually on standalone once.
